### ISSUE TITLE
Request!

### ISSUE THREAD

**Issue by Sandrituky** [AMGarkin/BDOcrypt#11]
This is not a issue.. but... could you make a program that decrypt and encrypt the new composed music files? They have no extension.

Their default location should be Documents > Music > songname

**Comment by AMGarkin** [comment]
I quit BDO long time ago, so I don't have those files. Can you share one of them so I can check how they are encrypted?

**Comment by Sandrituky** [comment]
Ohh, you should come back, there are a lot of new things lately and the game is being generous lately, haha. 
[Music file.zip](https://github.com/AMGarkin/BDOcrypt/files/4663280/Music.file.zip)
I didn't know if you even would read this, thank you a lot for checking it ^^

**Comment by AMGarkin** [comment]
That's shai music file, right? I'm not really sure if it is really encrypted, my guess is that it contains just information about notes (something like midi file). To figure out format of this file I'd need to install game, create shai alt character and then experiment - save a few notes, check saved file...
Sorry, I won't help you with this file.

**Comment by Sandrituky** [comment]
Yeah, it doesn't have any extension. It should be totally like a midi file, it stores the same information. 
This is a image from the game: 
![image](https://user-images.githubusercontent.com/15177276/82590629-4ef46900-9b9e-11ea-90fc-39ccce491472.png)
I totally understand it, don't worry ^^ 
A file with no extension just breaks my mind, I wonder how could I read and edit this file, or convert to midi, and reconvert it to bdo music. If you have any idea, just tell me the technology or thing it could work and I'll try to investigate myself. 
Thanks for your reply!

**Comment by Sandrituky** [comment]
Well, don't worry xD I've tried to do what you said, and I saw the lines that changed, and... what I thought it was an instrument volume, it could also be the duration of a note of a different instrument. I'll keep manually copypasting from midi to game XD This is too complicated to my head. Thank you for having read me ^^

**Comment by isitfreetho** [comment]
> 
> 
> Well, don't worry xD I've tried to do what you said, and I saw the lines that changed, and... what I thought it was an instrument volume, it could also be the duration of a note of a different instrument. I'll keep manually copypasting from midi to game XD This is too complicated to my head. Thank you for having read me ^^

Have you made any progress with this at all? I am super interested in converting MIDI to BDO music files!

I'm an amateur and don't really fully know any coding languages, but I'm able to figure things out via google search and fiddling with code to get things to somewhat do what I need them to. I tried inputting the contents of the file into different language detectors, but it seems they always fail because of an unknown artifact. I'm assuming its because I don't have the characters for the specific korean characters? or they're specific for BDO?

**Comment by Sandrituky** [comment]
Hello. I'm sorry to tell you that I haven't made any progress... I'm handmade copy pasting midi notes in FL Studio to BDO player. I think the file is encrypted... I don't know :S



________________________________
De: isitfreetho ***@***.***>
Enviado: lunes, 22 de marzo de 2021 2:16
Para: AMGarkin/BDOcrypt ***@***.***>
Cc: Sandrituky ***@***.***>; Author ***@***.***>
Asunto: Re: [AMGarkin/BDOcrypt] Request! (#11)


Well, don't worry xD I've tried to do what you said, and I saw the lines that changed, and... what I thought it was an instrument volume, it could also be the duration of a note of a different instrument. I'll keep manually copypasting from midi to game XD This is too complicated to my head. Thank you for having read me ^^

Have you made any progress with this at all? I am super interested in converting MIDI to BDO music files!

I'm an amateur and don't really fully know any coding languages, but I'm able to figure things out via google search and fiddling with code to get things to somewhat do what I need them to. I tried inputting the contents of the file into different language detectors, but it seems they always fail because of an unknown artifact. I'm assuming its because I don't have the characters for the specific korean characters? or they're specific for BDO?

**Comment by Farliam93** [comment]
Hello, I also wanted to program a MIDI to MusicFile converter. I was able to find the files for the composer; they are in JS code and refer to internal C/++ codes via LUA. However, the pure notes seem to me to be simple XML. Therefore, these XML files must be encrypted by some function. I will use IDA to see if I can find any other functions in the internal client.

<img width="1571" height="752" alt="Image" src="https://github.com/user-attachments/assets/3d16278e-9487-4ebb-abb1-7a5a6b63c3bf" />
